src/modules/users/users.service.ts

```diff
@@ -21,12 +20,12 @@ export class UsersService {
         return await this.usersRepository.save(user);
     }
 
-    async findOneById(id: string): Promise<User | undefined> {
-        return this.usersRepository.findOne({where: {id: id}});
+    async findOneById(id: string): Promise<User> {
+        return await this.usersRepository.findOneOrFail({ where: { id: id } });
```

Comment:
just for knowledge:
-  https://eslint.org/docs/latest/rules/no-return-await
- https://jakearchibald.com/2017/await-vs-return-vs-return-await/

Reply:
I've removed the return await